docs/input processing.md

```diff
@@ -0,0 +1,40 @@
+# Input processing
+The input process is about processing any input into our internal model representation `CommonInputModel`. This most likely have to be done using different methods based on which kind of input it is.
+
+As of now two inputs are supported:
+- JSON Schema Draft 7
+- AsyncAPI version 2.0.0
```

Comment:
Is it AsyncAPI (the whole file) or AsyncAPI Schema Object?

Reply:
The whole file. At the moment you cannot say `.payload()` directly, you need the underlying JSON schema for it to be processed - `.payload()._json`

Reply:
@fmvilas In the long run I would like for us to be able to provide it with any input from the parsed AsyncAPI document and it would generate all the models for that. Example operation, message, definition etc.

Reply:
That makes sense. So then, from an AsyncAPI document, it generates all the models?

Reply:
Yes, it goes through all defined messages and generates those - https://github.com/asyncapi/generator-model-sdk/blob/6480dce4649750e83118ff74172f7ce5aee0bc27/src/processors/AsyncAPIInputProcessor.ts#L23